catalogue/webapp/components/SearchPageLayout/SearchPageLayout.tsx

```diff
@@ -163,28 +166,19 @@ const SearchLayout: FunctionComponent<{ hasEventsExhibitions: boolean }> = ({
 
   const updateUrl = (form: HTMLFormElement) => {
     const formValues = formDataAsUrlQuery(form);
-    if (formValues.query) {
-      const sortOptionValue = formValues?.sortOrder
-        ? Array.isArray(formValues.sortOrder)
-          ? formValues.sortOrder[0]
-          : formValues.sortOrder
-        : '';
 
-      const { sort, sortOrder } = getUrlQueryFromSortValue(sortOptionValue);
+    const sortOptionValue = getQueryPropertyValue(formValues.sortOrder);
+    const urlFormattedSort =
+      sortOptionValue && getUrlQueryFromSortValue(sortOptionValue);
 
-      const link = linkResolver({
-        ...formValues,
-        sortOrder,
-        sort,
-      });
-
-      return router.push(link.href, link.as);
-    } else {
-      router.push({
-        pathname: router.pathname,
-        query: {},
-      });
-    }
+    const link = linkResolver({
+      ...formValues,
+      ...(urlFormattedSort && {
+        sort: urlFormattedSort.sort,
+        sortOrder: urlFormattedSort.sortOrder,
+      }),
```

Comment:
It's not immediately clear to me why we need to both spread `urlFormattedSort` and specify both values; couldn't get do just one?

Reply:
If you're asking why we have `sort` and `sortOrder`, I believe it was to keep the same URL structure as before and match the Javascriptless form as well.